### PR TITLE
feat: Add CRUD for Asana tasks

### PR DESCRIPTION
**Pull request by martinalbert**
## Description
Adds new usecases and maps under profile `project-management/tasks`:
- `ReadAllTasks`,
- `ReadTask`
- `UpdateTask`
- `RemoveTask`

## Motivation and Context
CRM features

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jnv** [comment]
@martinalbert Is this ready? I see the tests are failing.

**Comment by jnv** [comment]
Just from a quick glance:

- I don't see a version bump of the profile, this is (seems?) like a minor change (or are there any breakages?)
- `ReadAllTasks` – I think `ListTasks` is preferred?

**Review by janhalama — COMMENTED** [review]
I went through it. And I have some suggestions and improvements.

File: grid/project-management/tasks/profile.supr [hunk 1]
```diff
@@ -42,31 +42,198 @@ usecase CreateTask {
...
 
-	error {
-		"Human-readable error message"
-		message! string!
-		"Human-readable details about the error"
-		description! string!
+"""
+List all tasks in a project
+"""
+usecase ListTasks {
```
[the remark this answers is not in the document]
Comment (tmladek) [reply grid/project-management/tasks/profile.supr:66]: What do you mean? Seems to follow the rest of station, with no missing indents

[the remark this answers is not in the document]
Comment (tmladek) [reply grid/project-management/tasks/profile.supr:66]: Oh I see it now 😄 @martinalbert used tabs, which in my VS code config looks just fine; the rest of the file uses spaces

[the remark this answers is not in the document]
Comment (martinalbert) [reply grid/project-management/tasks/profile.supr:66]: It's more visible in body of usecases 😄 I also noticed this and vscode won't help for some reason, I'll fix it.

[the remark this answers is not in the document]
Comment (martinalbert) [reply grid/project-management/tasks/profile.supr:66]: @janhalama @tmladek should be better now 🙌🏻

File: grid/project-management/tasks/maps/asana.suma [hunk 2]
```diff
@@ -22,10 +18,193 @@ map CreateTask {
...
+    security "personalAccessToken"
+    
+    response 200 "application/json" {
+      data = {
+        id: body.data.gid,
+        title: body.data.name,
+        description: body.data.notes,
+        projects: body.data.projects.map(project => project.gid),
+        parent: body.data.parent,
+        assignee: body.data.assignee.gid,
```
Comment (janhalama) [inline grid/project-management/tasks/maps/asana.suma:82]: issue: assignee is nullable

File: grid/project-management/tasks/maps/asana.suma [hunk 3]
```diff
@@ -22,10 +18,193 @@ map CreateTask {
...
+    }
+  }
+}
+
+map ReadTask {
+  http GET "/tasks/{input.id}" {
+    security "personalAccessToken"
+    
+    response 200 "application/json" {
+      data = {
```
Comment (janhalama) [inline grid/project-management/tasks/maps/asana.suma:76]: suggestion(non-blocking): mapping of a task in `CreateTask`, `UpdateTask` and `ReadTask` use cases contains the same code I suggest to refactor the code and move the mapping into operation

File: grid/project-management/tasks/maps/asana.suma [hunk 4]
```diff
@@ -22,10 +18,193 @@ map CreateTask {
...
+map UpdateTask {
+  http PUT "/tasks/{input.id}" {
+    security "personalAccessToken"
+    
+    request "application/json" {
+      body {
+        data = {
+          name: input.title,
+          notes: input.description,
+          // projects: [input.project],
```
Comment (janhalama) [inline grid/project-management/tasks/maps/asana.suma:110]: issue: you have commented out code here without any comment, IMO the commented out properties should be updatable

File: grid/project-management/tasks/maps/asana.suma [hunk 5]
```diff
@@ -22,10 +18,193 @@ map CreateTask {
...
     }
+    
+    response 200 "application/json" {
+      data = {
+        id: body.data.gid,
+        title: body.data.name,
+        description: body.data.notes,
+        projects: body.data.projects.map(project => project.gid),
+        parent: body.data.parent,
+        assignee: body.data.assignee.gid,
```
Comment (janhalama) [inline grid/project-management/tasks/maps/asana.suma:124]: issue: dtto assignee is nullable

File: grid/project-management/tasks/profile.supr (hunk 1 above)
[the remark this answers is not in the document]
Comment (janhalama) [reply grid/project-management/tasks/profile.supr:66]: LGTM 🙏

**Review by janhalama — CHANGES_REQUESTED** [review]
I found some issues in the map.

**Review by janhalama — APPROVED** [review]
Looks good 👍

**Review by tmladek — APPROVED** [review]
lgtm, feel free to merge

**Pull request merged**